helix-core/src/main/java/org/apache/helix/api/listeners/CustomizedStateRootChangeListener.java

```diff
@@ -0,0 +1,36 @@
+package org.apache.helix.api.listeners;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.NotificationContext;
+
+
+/**
+ * Interface to implement to respond to changes in the root path of customized state
+ */
+public interface CustomizedStateRootChangeListener {
+
+  /**
+   * Invoked when root path customized state changes
+   * @param instanceName name of the instance whose state changed
+   * @param changeContext the change event and state
+   */
+  void onCustomizedStateRootChange(String instanceName, NotificationContext changeContext);
```

Comment:
Shall this method be passed a list of types in case anyone defined it with PreFetch == true?